internal/common/db.go

```diff
@@ -171,6 +171,16 @@ func GetHostByIdFromDB(db *gorm.DB, hostId string) (*Host, error) {
 	return &host, nil
 }
 
+func GetHostFromDBWhere(db *gorm.DB, where ...interface{}) (*Host, error) {
```

Comment:
you should also remove the `GetHostByIdFromDB` function, as it was only used by `GetHostById` that you did (rightfully) removed.

Reply:
actually, you do not have `hostId`. looking again